src/packages/core/property-editor/uis/tree-picker/property-editor-ui-tree-picker.element.ts

```diff
@@ -52,6 +62,35 @@ export class UmbPropertyEditorUITreePickerElement extends UmbLitElement implemen
 		this.ignoreUserStartNodes = config?.getValueByAlias('ignoreUserStartNodes');
 	}
 
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_WORKSPACE_CONTEXT, (workspaceContext) => {
+			this.#workspaceContext = workspaceContext;
+		});
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		this.#setStartNodeId();
+	}
+
+	#setStartNodeId() {
+		if (this.startNodeId) return;
+
+		const entityId = this.#workspaceContext?.getEntityId();
+		// TODO: Awaiting the workspace context to have a parent entity ID value. [LK]
+		// e.g. const parentEntityId = this.#workspaceContext?.getParentEntityId();
+		if (entityId && this.#dynamicRoot) {
+			this.#dynamicRootRepository.postDynamicRootQuery(this.#dynamicRoot, entityId).then((result) => {
+				if (result) {
+					this.startNodeId = result[0];
+				}
```

Comment:
I would really like if we could double wrap this check to ensure `result` has a length before accessing the 0-index.

Reply:
Good spot, I've added the extra check now.